python/ray/serve/handle.py

```diff
@@ -637,9 +637,12 @@ async def __call__(self, start: int) -> int:
 
     def __await__(self):
         """Yields the final result of the deployment handle call."""
-        obj_ref = yield from asyncio.wrap_future(self._object_ref_future)
-        result = yield from obj_ref.__await__()
-        return result
+        try:
+            obj_ref = yield from asyncio.wrap_future(self._object_ref_future)
+            result = yield from obj_ref.__await__()
+            return result
+        except ray.exceptions.TaskCancelledError:
```

Comment:
This is just to catch unhandled error during the cancel call. If we think this should be raise, I can revert this change.

Reply:
This should still be raised, otherwise the API is very strange/unexpected: if you cancel a ref, then `await` the result, you get back `None`. This would be indistinguishable from a handler running successfully and returning `None`

Reply:
Sounds good, I just feel it might confuse users seeing an unhandled error in the log when cancel happened. But I can also see that it's expected to see such log when the client just dropped. Will put it back